### PR TITLE
refacto: pass new CI

### PR DESCRIPTION
**Pull request by jbcaron**
# Pull Request type

- Refactoring (no functional changes, no API changes)

## What is the new behavior?

- Removal of unnecessary `into()` calls
- Removal of unnecessary `clone()` calls and dereferencing when necessary
- Change implementations `Into` into to `From` when not bidirectional
- add `./.markdownlint.json' for CI

## Does this introduce a breaking change?

🤷‍♂️

### CODE REVIEW

**Review by antiyro — APPROVED** [review]

File: crates/client/sync/src/l1.rs [hunk 1]
```diff
@@ -207,8 +204,7 @@ pub fn update_l1(state_update: L1StateUpdate) {
 
 /// Verify the L1 state with the latest data
 pub async fn verify_l1(state_update: L1StateUpdate, rpc_port: u16) -> Result<(), String> {
-    let starknet_state = STARKNET_STATE_UPDATE.lock().map_err(|e| e.to_string())?;
-    let starknet_state_block_number = starknet_state.block_number;
+    let starknet_state_block_number = STARKNET_STATE_UPDATE.lock().map_err(|e| e.to_string())?.block_number;
```
Comment (Trantorian1) [inline crates/client/sync/src/l1.rs:207]: Maybe replace `map_err(|e| e.to_string())` with a reference to `to_string` in the map.

File: crates/primitives/contract/src/class.rs [hunk 2]
```diff
@@ -191,19 +189,19 @@ pub mod convert {
...
+    impl From<AbiEventEntryWrapper> for LegacyEventAbiEntry {
+        fn from(abi_event_entry: AbiEventEntryWrapper) -> Self {
             LegacyEventAbiEntry {
-                r#type: self.r#type.into(),
-                name: self.name,
-                keys: self.keys.into_iter().map(|v| v.into()).collect(),
-                data: self.data.into_iter().map(|v| v.into()).collect(),
+                r#type: abi_event_entry.r#type.into(),
+                name: abi_event_entry.name,
+                keys: abi_event_entry.keys.into_iter().map(|v| v.into()).collect(),
```
Comment (Trantorian1) [inline crates/primitives/contract/src/class.rs:203]: Could you please replace `|v| v.into()` with a reference to the associated `from` function?

File: crates/primitives/contract/src/class.rs [hunk 3]
```diff
@@ -232,18 +230,18 @@ pub mod convert {
...
+        fn from(abi_struct_entry: AbiStructEntryWrapper) -> Self {
             LegacyStructAbiEntry {
-                r#type: self.r#type.into(),
-                name: self.name,
-                size: self.size,
-                members: self.members.into_iter().map(|v| v.into()).collect(),
+                r#type: abi_struct_entry.r#type.into(),
+                name: abi_struct_entry.name,
+                size: abi_struct_entry.size,
+                members: abi_struct_entry.members.into_iter().map(|v| v.into()).collect(),
```
Comment (Trantorian1) [inline crates/primitives/contract/src/class.rs]: Same thing here, replace calls to `v.into()`

File: crates/primitives/transactions/src/from_broadcasted_transactions.rs [hunk 4]
```diff
@@ -113,10 +113,10 @@ impl TryFrom<BroadcastedDeclareTransaction> for UserTransaction {
...
                         program: serde_json::from_slice(decompressed_bytes.as_slice())
                             .map_err(|_| BroadcastedTransactionConversionError::ProgramDeserializationFailed)?,
-                        abi: match contract_class.abi.as_ref() {
-                            Some(abi) => Some(abi.iter().cloned().map(|entry| entry.into()).collect()),
-                            None => None,
-                        },
+                        abi: contract_class
+                            .abi
+                            .as_ref()
+                            .map(|abi| abi.iter().cloned().map(|entry| entry.into()).collect()),
```
Comment (Trantorian1) [inline crates/primitives/transactions/src/from_broadcasted_transactions.rs:119]: Please replace `entry.into()` with a reference to the associated `from` or `into` function.

**Review by Trantorian1 — CHANGES_REQUESTED** [review]
Please fix the above issues.

File: crates/primitives/contract/src/class.rs (hunk 3 above)
Comment (jbcaron) [reply crates/primitives/contract/src/class.rs]: use `from` implemention

Comment (jbcaron) [reply crates/primitives/contract/src/class.rs]: not possible

**Pull request merged**